src/state/routing/slice.ts

```diff
@@ -133,21 +140,41 @@ export const routingApi = createApi({
               amount,
               type,
             })
-            return (await fetch(`quote?${query}`)) as { data: GetQuoteResult } | { error: FetchBaseQueryError }
-          } else {
-            const router = getRouter(args.tokenInChainId)
-            return await getClientSideQuote(
-              args,
-              router,
-              // TODO(zzmp): Use PRICE_PARAMS for RouterPreference.PRICE.
-              // This change is intentionally being deferred to first see what effect router caching has.
-              CLIENT_PARAMS
+            const response = await fetch(`quote?${query}`)
+            if (response.error) {
+              try {
+                // cast as any here because we do a runtime check on it being an object before indexing into .errorCode
+                const errorData = response.error.data as any
+                // NO_ROUTE should be treated as a valid response to prevent retries.
+                if (typeof errorData === 'object' && errorData?.errorCode === 'NO_ROUTE') {
```

Comment:
nit: Do we have any other types of errors? Are we expecting to have more in the future? If yes to any of these, maybe good opportunity for enum?